compiler/hash-typecheck/src/storage/primitives.rs

```diff
@@ -110,14 +118,19 @@ impl Scope {
 
     /// Add a member to the scope, overwriting any existing member with the same
     /// name.
-    pub fn add(&mut self, member: Member) {
+    pub fn add(&mut self, member: Member) -> usize {
         self.members.push(member);
-        self.member_names.insert(member.name, self.members.len() - 1);
+        let index = self.members.len() - 1;
+
+        self.member_names.insert(member.name, index);
+        index
     }
 
     /// Get a member by name.
-    pub fn get(&self, member_name: Identifier) -> Option<Member> {
-        Some(self.members[self.member_names.get(&member_name).copied()?])
+    pub fn get(&self, member_name: Identifier) -> Option<(Member, usize)> {
```

Comment:
Can return a scope member here.

Reply:
There is no access scope_id which ScopeMember requires